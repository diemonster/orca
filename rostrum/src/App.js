import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from './store/configureStore';

import KubeNav from './components/KubeNav';
import Namespace from './components/namespaces/Namespace';
import Pod from './components/pods/Pod';

const initialState = {
  namespace: {
    namespaceObjects: [],
    namespaceCreateInput: '',
    namespaceDeleteInput: '',
  },
  rolebinding: {
    rolebindingObjects: {},
  },
};

const store = configureStore(initialState);

function App() {
  return (
    <Provider store={store}>
      <Router>
        <div className="app-container">
          <h1>
            Orca: The Kubernetes Permissions Manager
          </h1>
          <div className='kube-container'>
            <KubeNav />
            <Switch>
              <Route exact path='/namespace/' component={Namespace} />
              <Route exact path='/pod/' component={Pod} />
            </Switch>
          </div>
        </div>
      </Router>
    </Provider>
  );
}

export default App;
