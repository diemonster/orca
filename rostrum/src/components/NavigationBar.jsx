import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { login, logout, setUsername } from '../actions/authActions';
import './navigation-bar.css';


export class NavigationBar extends React.Component {
  constructor(props) {
    super(props);

    this.handleLogin = this.handleLogin.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogin() {
    const { dispatchLogin } = this.props;
    dispatchLogin();
  }

  handleLogout() {
    const { dispatchLogout } = this.props;
    dispatchLogout();
  }

  render() {
    const { dispatchSetUsername, isAuthenticated, username } = this.props;

    if (isAuthenticated) {
      return (
        <div className="navigation-bar">
          <ul>
            <li className="profile">
              {username}
            </li>
            <li className="auth">
              <button type="button" onClick={this.handleLogout}>
              Log Out
              </button>
            </li>
          </ul>
        </div>
      );
    }

    if (username) {
      dispatchSetUsername('');
    }

    return (
      <div className="navigation-bar">
        <ul>
          <li className="auth">
            <button type="button" onClick={this.handleLogin}>
            Log In
            </button>
          </li>
        </ul>
      </div>
    );
  }
}

NavigationBar.propTypes = {
  dispatchLogin: PropTypes.func.isRequired,
  dispatchLogout: PropTypes.func.isRequired,
  dispatchSetUsername: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  username: state.auth.username,
});

const mapDispatchToProps = dispatch => ({
  dispatchLogin: () => dispatch(login()),
  dispatchLogout: () => dispatch(logout()),
  dispatchSetUsername: username => dispatch(setUsername(username)),
});

export default connect(mapStateToProps, mapDispatchToProps)(NavigationBar);
