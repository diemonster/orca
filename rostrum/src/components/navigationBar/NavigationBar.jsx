import React from 'react';
import './navigation-bar.css';
import PropTypes from 'prop-types';

function NavigationBar(props) {
  const { login, logout, name } = props;
  if (logout) {
    return (
      <div className="navigation-bar">
        <ul>
          <li className="home"><a href="/">Home</a></li>
          <li className="profile">
            <button type="button" onClick={logout}>
              {name ? `${name}, Log Out` : 'Log Out'}
            </button>
          </li>
        </ul>
      </div>
    );
  }

  return (
    <div className="navigation-bar">
      <ul>
        <li className="profile">
          <button type="button" onClick={login}>
            Log In
          </button>
        </li>
      </ul>
    </div>
  );
}

NavigationBar.propTypes = {
  login: PropTypes.func,
  logout: PropTypes.func,
  name: PropTypes.string,
};

NavigationBar.defaultProps = {
  login: null,
  logout: null,
  name: null,
};

export default NavigationBar;
