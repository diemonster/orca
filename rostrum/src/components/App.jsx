import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { handleAuthentication } from '../actions/authActions';
import NavigationBar from './NavigationBar';
import Namespaces from './Namespaces';
import Rolebindings from './Rolebindings';


export class App extends React.Component {
  componentWillMount() {
    this.handleAuthentication();
  }

  // If the URL has been populated with information from logging in through
  // Auth0, then call the authenticator's handleAuthentication method to
  // set session information
  handleAuthentication() {
    const { dispatchHandleAuthentication, location } = this.props;
    if (/access_token|id_token|error/.test(location.hash)) {
      dispatchHandleAuthentication(location.hash);
    }
  }

  render() {
    const { isAuthenticated } = this.props;

    if (isAuthenticated) {
      return (
        <div className="app-container">
          <NavigationBar />
          <h1>Orca: the Kubernetes permissions manager</h1>
          <div className="kube-container">
            <Namespaces />
            <Rolebindings />
          </div>
        </div>
      );
    }

    return (
      <div className="app-container">
        <NavigationBar />
        <h1>Orca: the Kubernetes permissions manager</h1>
      </div>
    );
  }
}

App.propTypes = {
  dispatchHandleAuthentication: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    hash: PropTypes.string.isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
});

const mapDispatchToProps = dispatch => ({
  dispatchHandleAuthentication: urlHash => dispatch(handleAuthentication(urlHash)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
