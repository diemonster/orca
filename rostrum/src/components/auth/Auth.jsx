import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';

import './auth.css';
import Authenticator from '../../authenticator/authenticator';

import App from '../App';
import NavigationBar from '../navigationBar/NavigationBar';

class Auth extends React.Component {
  componentWillMount() {
    this.setState({ profile: {} });

    const cb = (err, profile) => {
      if (err) {
        toastr.error(err);
      } else {
        this.setState({ profile });
      }
    };

    const { authenticator } = this.props;
    authenticator.getProfile(cb);
  }

  render() {
    const { authenticator } = this.props;
    const { profile } = this.state;
    if (authenticator.isAuthenticated()) {
      return (
        <div className="home-container">
          <NavigationBar name={profile.name} logout={authenticator.logout} />
          <App />
        </div>
      );
    }

    return (
      <div className="login-container">
        <NavigationBar login={authenticator.login} />
        <h1>ORCA</h1>
      </div>
    );
  }
}

Auth.propTypes = {
  authenticator: PropTypes.instanceOf(Authenticator).isRequired,
};

export default Auth;
