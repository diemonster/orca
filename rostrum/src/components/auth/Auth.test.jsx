import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Auth from './Auth';
import Authenticator from '../../authenticator/authenticator';
import NavigationBar from '../navigationBar/NavigationBar';
import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('Auth', () => {
    it('should render a login view when unauthenticated', () => {
      const authenticator = new Authenticator('', '');
      authenticator.login = jest.fn();
      authenticator.isAuthenticated = jest.fn(() => false);
      authenticator.getProfile = jest.fn((cb) => { cb(null, {}); });

      const props = { authenticator };

      const auth = shallow(<Auth {...props} />);

      expect(auth).toMatchSnapshot();
      expect(auth.find('div').hasClass('login-container')).toBe(true);
      expect(auth.find(NavigationBar).prop('login')).toBe(authenticator.login);
      expect(auth.find('h1').text()).toEqual('ORCA');
    });

    it('should render a logout view when authenticated', () => {
      const authenticator = new Authenticator('', '');
      authenticator.logout = jest.fn();
      authenticator.isAuthenticated = jest.fn(() => true);
      authenticator.getProfile = jest.fn((cb) => {
        cb(null, { name: 'username' });
      });

      const props = { authenticator };

      const auth = shallow(<Auth {...props} />);

      expect(auth).toMatchSnapshot();
      expect(auth.find('div').hasClass('home-container')).toBe(true);
      expect(auth.find(NavigationBar).prop('name')).toBe('username');
      expect(auth.find(NavigationBar).prop('logout')).toBe(authenticator.logout);
      expect(auth.find(App).exists()).toBe(true);
    });
  });
});
