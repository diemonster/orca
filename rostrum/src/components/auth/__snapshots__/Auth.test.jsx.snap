// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Auth should render a login view when unauthenticated 1`] = `
<div
  className="login-container"
>
  <NavigationBar
    login={[Function]}
    logout={null}
    name={null}
  />
  <h1>
    ORCA
  </h1>
</div>
`;

exports[`components Auth should render a logout view when authenticated 1`] = `
<div
  className="home-container"
>
  <NavigationBar
    login={null}
    logout={[Function]}
    name="username"
  />
  <App />
</div>
`;
